man/systemd.network.xml

```diff
@@ -1681,7 +1681,15 @@
         <varlistentry>
         <term><varname>DenyList=</varname></term>
         <listitem>
-          <para>A whitespace-separated list of IPv4 addresses. DHCP offers from servers in the list are rejected.</para>
+          <para>A whitespace-separated list of IPv4 addresses. DHCP offers from servers in the list are rejected. Note that
+          if <varname>AllowList=</varname> is configured then then <varname>DenyList=</varname> is ignored.</para>
```

Comment:
"then then"

Those semantics sound strange: the other should be consulted too. Let's just follow the same semantics as for IPAddressAllow/Deny=. It makes sense to use the same rules for similar concepts, that way users can transfer knowledge between areas.

Reply:
I am not following you. Can you be more clear.

Reply:
You say that AllowList= completely overrides DenyList=, DenyList= is not even looked at if the other is present. I'm asking instead for the semantics to be like that for IPAddressDeny=, IPAddressAllow=. How the two lists interact is described in the man page.

Reply:
>Access will be granted in case an IP packet's destination/source address matches any entry in the IPAddressAllow= setting.
Otherwise, access will be denied in case its destination/source address matches any entry in the IPAddressDeny= setting.
Otherwise, access will be granted.

The DHCPv4 client only talks with only one server.  If the `allow-list` is configured then, what is not in there will automatically blocked. The use case here is user wants to select only one DHCP server and block all others.

Reply:
@keszybz  are you onboard with this ?

Reply:
I'm not entirely convinced, but let's do this like that.